oneof regex validator fails to compile

```proto
syntax = "proto3";
import "github.com/mwitkow/go-proto-validators/validator.proto";

message FileObject {
    oneof hashes {
        string sha256   = 1 [(validator.field) = {regex: "[0-9a-fA-F]{64}"}];
        //string md5      = 2 [(validator.field) = {regex: "[0-9a-fA-F]{32}"}];
    }
}
```

```
$ protoc -I /go/src -I . --gogofast_out=. --govalidators_out=gogoimport=true:. re_test.proto
$ go build re_test.pb.go re_test.validator.pb.go
./re_test.validator.pb.go:31:7: undefined: _regex_FileObject_Sha256
```

It looks like the generated regex var has the name of the `oneof` wrapper instead of the field itself.

```go
var _regex_FileObject_Hashes = regexp.MustCompile("[0-9a-fA-F]{64}")

func (this *FileObject) Validate() error {
	if oneOfNester, ok := this.GetHashes().(*FileObject_Sha256); ok {
		if !_regex_FileObject_Sha256.MatchString(oneOfNester.Sha256) {
			return go_proto_validators.FieldError("Sha256", fmt.Errorf(`value '%v' must be a string conforming to regex "[0-9a-fA-F]{64}"`, oneOfNester.Sha256))
		}
	}
	return nil
}
```

--- a/plugin/plugin.go
+++ b/plugin/plugin.go
@@ -147,7 +147,7 @@ func (p *plugin) generateRegexVars(file *generator.FileDescriptor, message *gene
 	for _, field := range message.Field {
 		validator := getFieldValidatorIfAny(field)
 		if validator != nil && validator.Regex != nil {
-			fieldName := p.GetFieldName(message, field)
+			fieldName := p.GetOneOfFieldName(message, field)
 			p.P(`var `, p.regexName(ccTypeName, fieldName), ` = `, p.regexPkg.Use(), `.MustCompile(`, "`", *validator.Regex, "`", `)`)
 		}
 	}

--- a/test/gogo/validator_test.go
+++ b/test/gogo/validator_test.go
@@ -486,3 +486,24 @@ func TestOneOf_Passes(t *testing.T) {
 	err := example.Validate()
 	assert.NoError(t, err, "This message should pass all validation")
 }
+
+func TestOneOf_Regex(t *testing.T) {
+	example := &OneOfMessage3{
+		SomeInt: 30,
+		Something: &OneOfMessage3_FiveRegex{
+			FiveRegex: "11", // fail
+		},
+	}
+	err := example.Validate()
+	assert.Error(t, err, "regex applied to oneof field should fail validation on FiveRegex")
+	assert.Contains(t, err.Error(), "FiveRegex", "error must err on the FiveRegex")
+
+	example = &OneOfMessage3{
+		SomeInt: 30,
+		Something: &OneOfMessage3_FiveRegex{
+			FiveRegex: "aaa", // pass
+		},
+	}
+	err = example.Validate()
+	assert.NoError(t, err, "This message should pass all validation")
+}

--- a/test/golang/validator_test.go
+++ b/test/golang/validator_test.go
@@ -510,3 +510,24 @@ func TestOneOf_Passes(t *testing.T) {
 	err := example.Validate()
 	assert.NoError(t, err, "This message should pass all validation")
 }
+
+func TestOneOf_Regex(t *testing.T) {
+	example := &OneOfMessage3{
+		SomeInt: 30,
+		Something: &OneOfMessage3_FiveRegex{
+			FiveRegex: "11", // fail
+		},
+	}
+	err := example.Validate()
+	assert.Error(t, err, "regex applied to oneof field should fail validation on FiveRegex")
+	assert.Contains(t, err.Error(), "FiveRegex", "error must err on the FiveRegex")
+
+	example = &OneOfMessage3{
+		SomeInt: 30,
+		Something: &OneOfMessage3_FiveRegex{
+			FiveRegex: "aaa", // pass
+		},
+	}
+	err = example.Validate()
+	assert.NoError(t, err, "This message should pass all validation")
+}

--- a/test/validator_proto3_oneof.proto
+++ b/test/validator_proto3_oneof.proto
@@ -24,5 +24,6 @@ message OneOfMessage3 {
   oneof something {
     uint32 three_int = 5 [(validator.field) = {int_gt: 20}];
     uint32 four_int = 6 [(validator.field) = {int_gt: 100}];
+    string five_regex = 7 [(validator.field) = {regex: "^[a-z]{2,5}$"}];
   }
 }

